src/Phinx/Db/Adapter/PostgresAdapter.php

```diff
@@ -234,25 +234,23 @@ public function createTable(Table $table, array $columns = [], array $indexes =
             $sql = rtrim($sql, ', '); // no primary keys
         }
 
-        $sql .= ');';
+        $sql .= ')';
+        $this->execute($sql);
 
         // process column comments
         if (!empty($this->columnsWithComments)) {
             foreach ($this->columnsWithComments as $column) {
-                $sql .= $this->getColumnCommentSqlDefinition($column, $table->getName());
+                $this->execute($this->getColumnCommentSqlDefinition($column, $table->getName()));
             }
         }
 
         // set the indexes
         if (!empty($indexes)) {
             foreach ($indexes as $index) {
-                $sql .= $this->getIndexSqlDefinition($index, $table->getName());
+                $this->execute($this->getIndexSqlDefinition($index, $table->getName()));
```

Comment:
This is not ok, one of the improvement made before is to not execute side-effects before all the logic has been run first. I'd suggest buffering the SQL into an array and then execute each of the entries in the array separately.

Reply:
Yeah, that makes a lot of sense. Fixed.